internal/xpkg/fetch.go

```diff
@@ -38,14 +44,46 @@ type Fetcher interface {
 type K8sFetcher struct {
 	client    kubernetes.Interface
 	namespace string
+	transport http.RoundTripper
 }
 
 // NewK8sFetcher creates a new K8sFetcher.
-func NewK8sFetcher(client kubernetes.Interface, namespace string) *K8sFetcher {
+func NewK8sFetcher(client kubernetes.Interface, namespace, caBundlePath string) (*K8sFetcher, error) {
```

Comment:
nitpick: since the `caBundlePath` is optional and I could also see additional optional features being added here, I wonder if it would make sense to do something like this (as an example):
```

type FetcherOpt func(k *K8sFetcher) error

func WithCustomCA(path string) FetcherOpt {
	return func(k *K8sFetcher) error {
		// Get the SystemCertPool, continue with an empty pool on error
		rootCAs, _ := x509.SystemCertPool()
		if rootCAs == nil {
			rootCAs = x509.NewCertPool()
		}

		// Read in the cert file
		certs, err := ioutil.ReadFile(filepath.Clean(path))
		if err != nil {
			return errors.Wrapf(err, "Failed to append %q to RootCAs", path)
		}

		// Append our cert to the system pool
		if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
			return errors.Errorf("No certificates could be parsed from %q", path)
		}

		// NOTE: could optionally just require that k.transport is
		// *http.Transport, but that would restrict adding WithRoundTripper
		// option in the future unless package was modified. Or just override
		// the transport entirely as you were already doing :)
		t, ok := k.transport.(*http.Transport)
		if !ok {
			return errors.New("Fetcher transport is not an HTTP transport")
		}
		t.TLSClientConfig = &tls.Config{RootCAs: rootCAs, MinVersion: tls.VersionTLS12}
		return nil
	}
}

// NewK8sFetcher creates a new K8sFetcher.
func NewK8sFetcher(client kubernetes.Interface, namespace string, opts ...FetcherOpt) (*K8sFetcher, error) {
	k := &K8sFetcher{
		client:    client,
		namespace: namespace,
		transport: http.DefaultTransport,
	}

	for _, o := range opts {
		if err := o(k); err != nil {
			return nil, err
		}
	}

	return k, nil
}
```
Would also allow for adding some unit tests to `NewK8sFetcher` (though they would be fairly trivial at this point) as well as potentially the `WithCustomCA` option if we wanted to pass in the system cert pool / additional certs instead of parsing them in function body (I imagine in that case we would want to parse the certs as early as possible, i.e. pass them down through setup functions, which would avoid parsing them for each controller that requires the k8s fetcher). Anyway, let me know if you have any thoughts on this pattern!